### ISSUE TITLE
Support pathlib for attachment files

### ISSUE THREAD

**Issue by zooba** [pytest-dev/pytest-nunit#23]
https://github.com/pytest-dev/pytest-nunit/blob/8e27275337af3216a3bddc16e793ee9637902361/pytest_nunit/plugin.py#L143-L147

Essentially, rewrite this to:

```python
 def add_attachment(self, file, description): 
     r = self.nunit_xml.cases[self.id] 
     r.setdefault('attachments', {})[os.fsdecode(file)] = description 
```

(Threw in the `setdefault` for fun - you don't have to keep it ;) )

There are likely other places that will need similar updating, but I haven't reviewed it all yet. I'd recommend `fsdecode` if it's available, as that will get you to a `str` on all platforms, but if you're still supported older versions you'll need a stub like:

```python
def fake_fsdecode(p):
    if isinstance(p, bytes_or_str_depending_on_python_version):
        return p.decode(os.getfsencoding())
    return str(p)
```

**Comment by adekanyetomie** [comment]
Hi
For clarification sake, am I to replace the older function with the newer one?

**Comment by zooba** [comment]
@adekanyetomie Yes, the suggested (but untested!) rewrite I put above will handle all the same cases as the original. But as I mention, it may need some extra work for older versions of Python.

**Comment by flying-sheep** [comment]
At this point, there’s no older versions of Python anymore that could cause problems here.
